extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/AuthenticationProviderService.java

```diff
@@ -306,6 +311,26 @@ public LDAPAuthenticatedUser authenticateUser(Credentials credentials)
         }
 
     }
+    /**
+     * Returns parameter current ldap domain token generated from user credentials
+     * If no multiple LDAP are configured on GUACAMOLE_HOME
+     * a null is returned.
+     *
+     * @param credentials
+     *     The credentials to use for authentication.
+     *
+     * @return
+     *     Domain name by splitting logged username when multiple LDAP configuration is available
+     *     or null if no such configuration
+     */
+    private String getDomainToken(Credentials credentials) {
+       String ldapDomainName = null;
+        // Creating custom LDAP attribute token - domain name - when configured to multiple LDAP
+        if (credentials.getUsername().contains("\\")) {
```

Comment:
I think we should probably deal with situations where the username is in the UPN format, which is `USERNAME`@`DOMAIN`?